File: Leetcode Challenge/March/C++/Is Subsequence.cpp
// class Solution {
// public:
//     bool isSubsequence(string s, string t) {
//         unordered_map<char, vector<int>> hg; 
//         for(int i=0; i<t.length();i++) hg[t[i]].push_back(i);
        
//         int prev = -1;
//         for(auto c : s){
//             auto it = hg.find(c);
//             if(it == hg.end()) return false;
//             auto vec = it->second;
//             int pos = upper_bound(vec.begin(), vec.end(), prev) - vec.begin(); 
//             if(pos == vec.size()) return false;
//             prev = vec[pos];
//         }
//         return true;
//     }
// };

class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s == ""){
            return true;
        }
        if(t == ""){
            return false;
        }    

        if(s.size() == 1 && t.size() == 1){
            return s[0] == t[0];
        }    

        int a = 0;
        for(int i=0; i<t.size();i++){
            
            if(s[a] == t[i]){
                a += 1;
            }
        }     
        
        
        if(a == s.size()){
            return true;
        }    
        else{
            return false;
        }    
    }
};

File: Leetcode Challenge/March/Python/Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        if s == "":
            return True
        if t == "":
            return False

        if len(s) == 1 and len(t) == 1:
            return s[0] == t[0]

        a = 0
        # idx=0
        b = 0
        for i in range(len(t)):
            # print(a,i)
            if a>=len(s):
                return True
            if s[a] == t[i]:

                a += 1

        # a += 1
        print(a)
        if a == len(s):
            return True
        else:
            return False
